### PR TITLE
Better UUID validation and NPE fix

### PR DESCRIPTION
**Pull request by radutudor**

### CODE REVIEW

**Review by kovacshuni — APPROVED** [review]

File: src/test/java/com/ft/methodecontentcollectionmapper/validation/ContentCollectionValidatorTest.java [hunk 1]
```diff
@@ -14,23 +16,84 @@
...
 
     @Before
     public void setUp() {
         contentCollectionValidator = new ContentCollectionValidator();
     }
 
+
+    @Test(expected = ValidationException.class)
+    public void thatNullMainUuidThrowsException() {
+        final EomFile eomFile = new EomFile(null, null, null, null, null, null, null, null);
```
Comment (bogdanguranda) [inline src/test/java/com/ft/methodecontentcollectionmapper/validation/ContentCollectionValidatorTest.java]: Does this class EomFile have a builder which you could use here with default values?

**Review by bogdanguranda — APPROVED** [review]

File: src/test/java/com/ft/methodecontentcollectionmapper/validation/ContentCollectionValidatorTest.java (hunk 1 above)
Comment (radutudor) [reply src/test/java/com/ft/methodecontentcollectionmapper/validation/ContentCollectionValidatorTest.java]: There's no builder, sadly...

We could add one, but we would be creating it exclusively for testing purposes, as the EomFile is never actually instantiated by us in the production code.

Should we add one?

Comment (kovacshuni) [reply src/test/java/com/ft/methodecontentcollectionmapper/validation/ContentCollectionValidatorTest.java]: yes, please

Comment (radutudor) [reply src/test/java/com/ft/methodecontentcollectionmapper/validation/ContentCollectionValidatorTest.java]: done

**Comment by kovacshuni** [comment]
awesome!! 💰

**Review by bogdanguranda — APPROVED** [review]
Thanks for adding Builders :+1:

File: pom.xml [hunk 2]
```diff
@@ -19,19 +19,22 @@
...
-		<ft-build-info-maven-plugin.version>1.0.2</ft-build-info-maven-plugin.version>
 		<maven-compiler-plugin.version>3.1</maven-compiler-plugin.version>
 		<maven-source-plugin.version>2.2.1</maven-source-plugin.version>
 		<maven-jar-plugin.version>2.3.2</maven-jar-plugin.version>
 		<surefire.plugin.version>2.18.1</surefire.plugin.version>
-		<jersey-version>1.18.1</jersey-version>
-		<resilientclient.version>0.4.0</resilientclient.version>
+		<coveralls-maven-plugin.version>4.3.0</coveralls-maven-plugin.version>
 		<target-jdk>1.8</target-jdk>
+		<argLine>-Djava.net.preferIPv4Stack=true</argLine>
```
Comment (tosan88) [inline pom.xml:37]: ah, nice!

**Review by tosan88 — APPROVED** [review]

File: src/test/java/com/ft/methodecontentcollectionmapper/validation/ContentCollectionValidatorTest.java (hunk 1 above)
Comment (tosan88) [reply src/test/java/com/ft/methodecontentcollectionmapper/validation/ContentCollectionValidatorTest.java]: cool, thanks!

**Pull request merged**